java/src/test/java/com/saucelabs/saucebindings/integration/VisualTest.java

```diff
@@ -0,0 +1,116 @@
+package com.saucelabs.saucebindings.integration;
+
+import com.saucelabs.saucebindings.DataCenter;
+import com.saucelabs.saucebindings.SauceSession;
+import com.saucelabs.saucebindings.SystemManager;
+import com.saucelabs.saucebindings.options.InvalidSauceOptionsArgumentException;
+import com.saucelabs.saucebindings.options.SauceOptions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class VisualTest {
+    private SauceSession session;
+    private RemoteWebDriver webDriver;
+
+    @Before
+    public void setup() {
+        System.setProperty("BUILD_NAME", "Sauce Bindings");
+    }
+
+    @After
+    public void cleanUp() {
+        if (session != null) {
+            session.stop(true);
+        }
+    }
+
+    @Test
+    public void usesVisualWhenOptionSet() {
+        SauceOptions options = SauceOptions.chrome().setName("Visual uses this name").setVisual().build();
+        session = new SauceSession(options);
+        webDriver = session.start();
+        session.visual().takeSnapshot("Snapshot necessary to for Visual test to pass");
+        assertNotNull(webDriver);
+        assertTrue(session.getIsVisualSession());
+    }
+
+    @Test
+    public void usesVisualWhenDataCenterSet() {
+        SauceOptions options = SauceOptions.chrome().setName("Visual uses this name").build();
+        session = new SauceSession(options);
+        session.setDataCenter(DataCenter.VISUAL);
+        webDriver = session.start();
+        session.visual().takeSnapshot("Snapshot necessary to for Visual test to pass");
+        assertNotNull(webDriver);
+        assertTrue(session.getIsVisualSession());
+    }
+
+    @Test
+    public void acceptsNameForInit() {
+        SauceOptions options = SauceOptions.chrome().setName("Visual does not use this name").setVisual().build();
+        session = new SauceSession(options);
+        webDriver = session.start();
+        session.visual().init("Visual uses this name instead");
+        session.visual().takeSnapshot("Snapshot necessary to for Visual test to pass");
+        Map results = session.visual().getResults();
```

Comment:
You mean convenience methods for parsing results?
Oh, we should make a `VisualResults` object and populate it in `getResults()` and then let the user query that rather than needing to parse the `Map`